src/lsp/sql_parser/lexer.mll

```diff
@@ -0,0 +1,170 @@
+   | "END" -> END
+   | "EXCEPT" -> EXCEPT
+   | "EXEC" -> EXEC
+   | "EXECUTE" -> EXECUTE
+   | "EXCEPTION" -> EXCEPTION
+   | "FETCH" -> FETCH
+   | "FOR" -> FOR
+   | "FOUND" -> FOUND
+   | "FROM" -> FROM
+   | "GOTO" -> GOTO
+   | "GO" -> GO
+   | "GROUP" -> GROUP
+   | "HAVING" -> HAVING
+   | "HOLD" -> HOLD
+   | "IDENTIFIED" -> IDENTIFIED
+   | "IGNORE" -> IGNORE
+   | "INNER" -> INNER
+   | "IMMEDIATE" -> IMMEDIATE
+   | "IN" -> IN
+   | "INCLUDE" -> INCLUDE
+   | "INSERT" -> INSERT
+   | "INTEGER" -> INTEGER
+   | "INTERSECT" -> INTERSECT
+   | "INTO" -> INTO
+   | "IS" -> IS
+   | "JOIN" -> JOIN
+   | "LEFT" -> LEFT
+   | "NATURAL" -> NATURAL
+   | "NOT" -> NOT
+   | "NULL" -> NULL
+   | "OF" -> OF
+   | "ON" -> ON
+   | "OR" -> OR
+   | "ORDER" -> ORDER
+   | "OPEN" -> OPEN
+   | "OUTER" -> OUTER
+   | "PERFORM" -> PERFORM
+   | "PREPARE" -> PREPARE
+   | "RAISE" -> RAISE
+   | "RELEASE" -> RELEASE
+   | "RESET" -> RESET
+   | "RIGHT" -> RIGHT
+   | "ROLLBACK" -> ROLLBACK
+   | "SAVEPOINT" -> SAVEPOINT
+   | "SECTION" -> SECTION
+   | "SELECT" -> SELECT
+   | "SET" -> SET
+   | "START" -> START
+   | "STATEMENT" -> STATEMENT
+   | "SQL" -> SQL
+   | "SQLERROR" -> SQLERROR
+   | "SQLWARNING" -> SQLWARNING
+   | "TABLE" -> TABLE
+   | "THEN" -> THEN
+   | "TIMESTAMP" -> TIMESTAMP
+   | "TO" -> TO
+   | "TRAN" -> TRANSACTION
+   | "TRANSACTION" -> TRANSACTION
+   | "UPDATE" -> UPDATE
+   | "UNION" -> UNION
+   | "USER" -> USER
+   | "USING" -> USING
+   | "VALUES" -> VALUES
+   | "VARCHAR" -> VARCHAR
+   | "WHEN" -> WHEN
+   | "WHERE" -> WHERE
+   | "WHENEVER" -> WHENEVER
+   | "WITH" -> WITH
+   | "WORK" -> WORK
+   | _ -> TOKEN x (* "Brut SQL" or sql variable, stay the same *)
+}
+
+rule token = parse
+  | "END-EXEC"
+      { END_EXEC }
+  | "end-exec"
+      { END_EXEC }
+  | "END-EXEC."
+      { END_EXEC }
+  | ':' (['a'-'z' 'A'-'Z']['A'-'Z' 'a'-'z' '0'-'9' '_' '-']* as s)
+      { COBOL_VAR s }
+  | '\\' (['a'-'z' 'A'-'Z']['A'-'Z' 'a'-'z' '0'-'9' '_']* as s)
+      { BACKSLASHED_VAR s }
+  | ['a'-'z' 'A'-'Z' '0'-'9']['A'-'Z' 'a'-'z' '0'-'9' '-' '_' '*' ]* as s 
+  (*If you want to make a better sql parsing, this is to rework*)
+      { get_keyword s }
+  | ['0'-'9']+ as s
+      { DIGITS s }
+  | '\'' ( ['A'-'Z' 'a'-'z' '0'-'9' '_' '(' '*' ')' '.' '[' ']']* as s) '\''
+      { STRING s}
+
+
+(*   | ['>' '<']+ as s (*sql opeatuin*)
+      { TOKEN s } *)
+  | "||"
+      { OR }
+  | ['='] 
+      { EQUAL }
+  | ['+'] 
+      { PLUS }
+  | ['-'] 
+      { MINUS }
+  | "<=" 
+      { LESS_EQ }
+  | ">=" 
+       { GREAT_EQ }
+  | "<" 
+        { LESS }
+  | ">" 
+        { GREAT } 
+  | ":" 
+        { COLON } 
+  | ','
+      { COMMA }
+  | '.'
+      { DOT }
+  | '('
+      { LPAR }
+  | ')'
+      { RPAR }
+  | '*'
+      { STAR }
+(*   | _ as s 
+      { TOKEN s } *)
+  | eof
+      { EOF }
+
+{
+  (* when empty lexbuf, we use an exeption... this is not very cool *)
+  let token lexbuf =
+    try token lexbuf with _ -> EOF
```

Comment:
Could you specify what exception is caught here? Also is there a way to avoid this case?

Reply:
Nothing, it was an artifact. I remove it